### PR TITLE
[CBRD-24016] It may fail to load Rootclass mop due to TDE

### PR DESCRIPTION
**Pull request by wrlawodms**
http://jira.cubrid.org/browse/CBRD-24016

### CODE REVIEW

File: src/transaction/locator_sr.c [hunk 1]
```diff
@@ -5347,12 +5347,16 @@ locator_update_force (THREAD_ENTRY * thread_p, HFID * hfid, OID * class_oid, OID
     {
       HEAP_OPERATION_CONTEXT update_context;
 
-      or_class_tde_algorithm (recdes, &tde_algo);
-      if (tde_algo != TDE_ALGORITHM_NONE && !tde_Cipher.is_loaded)
+      if (!OID_IS_ROOTOID (oid))
```
Comment (hornetmj) [inline src/transaction/locator_sr.c:5350]: How about checking it with class_oid together?

if (OID_IS_ROOTOID (class_oid) && !OID_IS_ROOTOID (oid))

Comment (wrlawodms) [reply src/transaction/locator_sr.c:5350]: `OID_IS_ROOTOID (class_oid)` is always true here (see 3 lines above). Do you think it is needed to make the condition clear?

**Review by ribram — APPROVED** [review]

File: src/transaction/locator_sr.c (hunk 1 above)
Comment (hornetmj) [reply src/transaction/locator_sr.c:5350]: I mean it looks good if the OID_IS_ROOTOID (class_oid) (3 lines above) and the !OID_IS_ROOTOID (oid) check together.

Comment (wrlawodms) [reply src/transaction/locator_sr.c:5350]: Other jobs have to be done even if `!OID_IS_ROOTOID (oid)` except for the TDE error handling.

**Review by hornetmj — APPROVED** [review]

File: src/transaction/locator_sr.c (hunk 1 above)
Comment (hornetmj) [reply src/transaction/locator_sr.c:5350]: Oh! I read the code incorrectly because of misreading of the indentation.

**Review by eido5 — APPROVED** [review]

**Review by joohok — APPROVED** [review]

**Comment by hornetmj** [comment]
@mhoh3963 We finished the review and we want to merge this PR. But the circleci complains of some testcase failure that may not be related to this PR. Could you please check it and merge it?

**Comment by mhoh3963** [comment]
@hornetmj  I merged it.

**Pull request merged**